Sources/LeafKit/LeafParser.swift

```diff
@@ -313,6 +323,10 @@ struct LeafParser {
         self.offset += 1
     }
     
+    private mutating func replace(at offset: Int = 0, with new: LeafToken) {
+        self.tokens[self.offset + offset] = new
+    }
+    
```

Comment:
Added to permit more intelligent parsing behavior to mutate the token stream in cases where it can determine an initial token is misclassified - eg, trailing colon on tags that expect no body (or in future, where a tagIndicator would inappropriately consumes HTML anchors or whatever, that tag can be determined not to exist and the token can revert to raw)